node_modules/gh-users/lib/internal/dao.js

```diff
@@ -145,20 +145,17 @@ var searchUsers = module.exports.searchUsers = function(appId, query, limit, off
     var sequelize = DB.getSequelize();
 
     // Construct a query that selects a set of rows in the Users table ordered
-    // on the sum of the displayName, email and shibbolethId similarities. Although
+    // on the sum of the displayName and shibbolethId similarities. Although
     // this isn't perfect, it gives a reasonable approximation of a typeahead search as
-    // each of the three columns their values are split up in possible trigrams and indexed.
-    // We need to coalesce the `email` and `shibbolethId` columns as you can't get a similarity
-    // value from a null value. Because of the index on each column, the query itself is
-    // reasonably quick and accurate
+    // each of the three columns their values are split up in possible trigrams and indexed
     var sqlQuery = 'SELECT "Users".* ';
     sqlQuery += 'FROM "Users" ';
-    sqlQuery += 'WHERE "AppId" = ? ';
-    sqlQuery += 'ORDER BY (similarity("displayName", ?) + similarity(coalesce("email", ?), ?) + similarity(coalesce("shibbolethId", ?), ?)) DESC, "displayName" ASC, "id" ASC ';
-    sqlQuery += 'LIMIT ? OFFSET ?' ;
-    var coalescedValue = '';
+    sqlQuery += 'WHERE "AppId" = ? and ("displayName" <-> ? < 0.8 or "shibbolethId" <-> ? < 0.8) ';
```

Comment:
I wouldn't be opposed to dropping the similarity search and just doing a more standard ilike match